presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -954,6 +954,30 @@ public static Block stEnvelopeAsPts(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return blockBuilder.build();
     }
 
+    @Description("Returns the bounding rectangle of a Geometry expanded by distance.")
+    @ScalarFunction("expand_envelope")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice expand_envelope(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(DOUBLE) double distance)
+    {
+        if (isNaN(distance)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "expand_envelope: distance is NaN");
+        }
+
+        if (distance < 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "expand_envelope: distance is negative");
```

Comment:
nit: perhaps, include the value of distance in the error message to simplify troubleshooting